pkg/compat/client.go

```diff
@@ -21,6 +21,7 @@ import (
 // A smart client.
 // Provides seamless API version compatibility.
 type Client struct {
+	*rest.Config
```

Comment:
It is beneficial to have config to be exposed in the custom client, as the client itself is short lived and non-cached, could be used in multiple occasion.